lib/pdc/resource/attributes.rb

```diff
@@ -90,15 +90,24 @@ def use_setters(attributes)
     end
 
     def method_missing(name, *args, &block)
-      if attribute?(name) then attribute(name)
+      if association?(name) then association(name).load
+      elsif attribute?(name) then attribute(name)
       elsif predicate?(name)   then predicate(name)
       elsif setter?(name)      then set_attribute(name, args.first)
       else super
```

Comment:
I met an error called "SystemStackError: stack level too deep" if write like this in belongs_to commit.
Maybe this association check should be after  setter?(name) , I have changed this in that pr.

Reply:
In my code, I haven't met this issue. Will try it with your codes later

Reply:
I think the problem is you haven't include necessary codes in spec/support/fixtures.rb, like following:
`   class CustomPrimaryKeyModel
@@ -37,6 +38,7 @@ class CustomPrimaryKeyModel
     include PDC::Resource::Attributes
     include PDC::Resource::Attributes
     include PDC::Resource::Identity
     include PDC::Resource::Identity
     include PDC::Resource::Scopes
     include PDC::Resource::Scopes
+    include PDC::Resource::Associations`

Reply:
You need to include PDC::Resource::Associations` for the tested model

Reply:
Cool, thx!